src/org/parosproxy/paros/core/scanner/VariantDdnPath.java

```diff
@@ -0,0 +1,109 @@
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.parosproxy.paros.core.scanner;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.model.SessionStructure;
+
+public class VariantDdnPath implements Variant {
+
+	private static final Logger LOGGER = Logger.getLogger(VariantDdnPath.class);
+	private final List<NameValuePair> stringParam = new ArrayList<>();
+
+	@Override
+	public void setMessage(HttpMessage msg) {
+		try {
+			List<String> treePaths = Model.getSingleton().getSession().getTreePath(msg);
+			int i = 0;
+			for (String path : treePaths) {
+				if (path.startsWith(SessionStructure.DATA_DRIVEN_NODE_PREFIX)) {
+					stringParam.add(new NameValuePair(NameValuePair.TYPE_URL_PATH, path, path, i+1));
+				}
+				i++;
+			}
+		} catch (URIException e) {
+			// Ignore
+		}
+	}
+
+	@Override
+	public List<NameValuePair> getParamList() {
+		return stringParam;
+	}
+
+	@Override
+	public String setParameter(HttpMessage msg, NameValuePair originalPair, String name, String value) {
+		return setParameter(msg, originalPair, name, value, false);
+	}
+
+	@Override
+	public String setEscapedParameter(HttpMessage msg, NameValuePair originalPair, String name, String value) {
+		return setParameter(msg, originalPair, name, value, true);
+	}
+
+	/**
+	 * Encode the parameter value for a correct URL introduction
+	 * 
+	 * @param value
+	 *            the value that need to be encoded
+	 * @return the Encoded value
+	 */
+	private String getEscapedValue(String value) {
+		if (value != null) {
+			try {
+				return URLEncoder.encode(value, "UTF-8");
+			} catch (UnsupportedEncodingException ex) {
+			}
+		}
+		return "";
+	}
+
+	private String setParameter(HttpMessage msg, NameValuePair originalPair, String name, String value,
+			boolean escaped) {
+		try {
+			URI uri = msg.getRequestHeader().getURI();
+			String[] paths = msg.getRequestHeader().getURI().getPath().toString().split("/");
+			if (originalPair.getPosition() < paths.length) {
+				String encodedValue = (escaped) ? value : getEscapedValue(value);
+				paths[originalPair.getPosition()] = encodedValue;
+				String path = StringUtils.join(paths, "/");
+				try {
+					uri.setEscapedPath(path);
+				} catch (URIException e) {
+					// Looks like it wasn't escaped after all

```

Comment:
Worth logging as warn or even error, if this happens it's because of a scanner misusing the variant (i.e. calling `setEscapedParameter` with a non-escaped value).